src/cli.rs

```diff
@@ -102,9 +117,62 @@ pub fn run_cli() {
     run_cli_with_args(std::env::args(), &GitOps);
 }
 
-fn extract_todos_from_files(files: &Vec<PathBuf>, marker_config: &MarkerConfig) -> Vec<MarkedItem> {
+/// Check if a file path should be excluded based on the exclude patterns.
+/// Returns true if the file should be excluded.
+fn should_exclude(file_path: &Path, exclude_paths: &[PathBuf]) -> bool {
+    for exclude_path in exclude_paths {
+        // Get the file name for simple file name matching
+        if let Some(file_name) = file_path.file_name() {
+            if let Some(exclude_name) = exclude_path.file_name() {
+                // Check if it's a simple file name match
+                if exclude_path.components().count() == 1 && file_name == exclude_name {
+                    return true;
+                }
+            }
+        }
+
+        // Normalize both paths
+        let file_normalized = file_path.components().collect::<PathBuf>();
+        let exclude_normalized = exclude_path.components().collect::<PathBuf>();
+
+        // Check if file path ends with the exclude pattern (handles relative paths)
+        if file_normalized.ends_with(&exclude_normalized) {
+            return true;
+        }
+
+        // Check if any component sequence matches
+        let file_components: Vec<_> = file_normalized.components().collect();
+        let exclude_components: Vec<_> = exclude_normalized.components().collect();
+
+        // Check if exclude pattern matches from any position in the file path
+        for i in 0..=file_components
+            .len()
+            .saturating_sub(exclude_components.len())
+        {
+            let slice: PathBuf = file_components[i..i + exclude_components.len()]
+                .iter()
+                .collect();
+            if slice == exclude_normalized {
+                return true;
+            }
+        }
+    }
+    false
+}
+
+fn extract_todos_from_files(
+    files: &Vec<PathBuf>,
+    marker_config: &MarkerConfig,
+    exclude_paths: &[PathBuf],
+) -> Vec<MarkedItem> {
     let mut new_todos = Vec::new();
     for file in files {
+        // Skip excluded files
+        if should_exclude(file, exclude_paths) {
+            info!("Excluding file: {:?}", file);
+            continue;
+        }
```

Comment:
Why are you checking for each file if they need to be excluded?

Why not filter the list of files before extracting the todos? 

So that files is already filtered